core/theme/zelos/classic.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Classic zelos theme.
+ */
+'use strict';
+
+goog.provide('Blockly.zelos.Themes.Classic');
+
+goog.require('Blockly.Themes.Classic');
+goog.require('Blockly.zelos.Theme');
+
+
+// Temporary holding object.
+Blockly.zelos.Themes.Classic = {};
+
+Blockly.zelos.Themes.Classic.defaultBlockStyles = {
+  "colour_blocks": {
+    "colourPrimary": "20"
+  },
+  "list_blocks": {
+    "colourPrimary": "260"
+  },
+  "logic_blocks": {
+    "colourPrimary": "210"
+  },
+  "loop_blocks": {
+    "colourPrimary": "120"
+  },
+  "math_blocks": {
+    "colourPrimary": "230"
+  },
+  "procedure_blocks": {
+    "colourPrimary": "290"
+  },
+  "text_blocks": {
+    "colourPrimary": "160"
+  },
+  "variable_blocks": {
+    "colourPrimary": "330"
+  },
+  "variable_dynamic_blocks": {
+    "colourPrimary": "310"
+  },
+  "hat_blocks": {
+    "colourPrimary": "330",
+    "hat": "cap"
+  }
+};
+
+Blockly.zelos.Themes.Classic.categoryStyles = {
+  "colour_category": {
+    "colour": "20"
+  },
+  "list_category": {
+    "colour": "260"
+  },
+  "logic_category": {
+    "colour": "210"
+  },
+  "loop_category": {
+    "colour": "120"
+  },
+  "math_category": {
+    "colour": "230"
+  },
+  "procedure_category": {
+    "colour": "290"
+  },
+  "text_category": {
+    "colour": "160"
+  },
+  "variable_category": {
+    "colour": "330"
+  },
+  "variable_dynamic_category": {
+    "colour": "310"
+  }
+};
+
+Blockly.zelos.Themes.Classic =
+    new Blockly.zelos.Theme('zelosclassic',
+        Blockly.zelos.Themes.Classic.defaultBlockStyles,
```

Comment:
Instead of having all of the block and category styles above, can you do
`new Blockly.zelos.Theme('zelosclassic', Blockly.Themes.Classic.defaultBlockStyles, Blockly.Themes.Classic.categoryStyles)`?

Reply:
No because Blockly.Themes.Classic.defaultBlockStyles is on a temporary holding object that gets replaced with the Blockly.Theme object when created. We could instead do ``Blockly.Themes.Classic.getAllBlockStyles()`` but that would have passed through the previous theme's validation methods which would have already generated the secondary / tertiary colours..